tests/contrib/celery/test_integration.py

```diff
@@ -195,6 +196,32 @@ def fn_exception():
         ok_('Traceback (most recent call last)' in span.get_tag('error.stack'))
         ok_('Task class is failing' in span.get_tag('error.stack'))
 
+    def test_fn_retry_exception(self):
+        # it should not catch retry exceptions in task functions
+        @self.app.task
+        def fn_exception():
+            raise Retry('Task class is being retried')
+
+        t = fn_exception.apply()
+        ok_(not t.failed())
+        ok_('Task class is being retried' in t.traceback)
+
+        traces = self.tracer.writer.pop_traces()
+        eq_(1, len(traces))
+        eq_(1, len(traces[0]))
+        span = traces[0][0]
+        eq_(span.name, 'celery.run')
+        eq_(span.resource, 'tests.contrib.celery.test_integration.fn_exception')
+        eq_(span.service, 'celery-worker')
+        eq_(span.get_tag('celery.id'), t.task_id)
+        eq_(span.get_tag('celery.action'), 'run')
+        eq_(span.get_tag('celery.state'), 'RETRY')
+
+        # TODO: these should be failing
```

Comment:
The reason these aren't failing is because you are explicitly raising a `Retry` exception.

This is something that Celery will catch and handle on it's own.

The other case we need to be able to support is when you set a retry policy for the task and another type of exception is raised.

```python
@self.app.task(autotry_for(Exception, ))
def fn_exception():
    raise Exception('oh no, what happened?!')
```